platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -916,32 +946,81 @@ int _odp_ipsec_sa_replay_precheck(ipsec_sa_t *ipsec_sa, uint32_t seq,
 				  odp_ipsec_op_status_t *status)
 {
 	/* Try to be as quick as possible, we will discard packets later */
-	if (ipsec_sa->antireplay &&
-	    seq + IPSEC_ANTIREPLAY_WS <=
-	    (odp_atomic_load_u64(&ipsec_sa->hot.in.antireplay) & 0xffffffff)) {
+	if (ipsec_sa->antireplay && ((seq + ipsec_sa->ar.win_size) <=
+	    (odp_atomic_load_u64(&ipsec_sa->hot.in.wintop_seq) & 0xffffffff))) {
 		status->error.antireplay = 1;
 		return -1;
 	}
 
 	return 0;
 }
 
-int _odp_ipsec_sa_replay_update(ipsec_sa_t *ipsec_sa, uint32_t seq,
-				odp_ipsec_op_status_t *status)
+static inline int ipsec_wslarge_replay_update(ipsec_sa_t *ipsec_sa, uint32_t seq,
+					      odp_ipsec_op_status_t *status)
 {
-	int cas = 0;
-	uint64_t state, new_state;
+	uint32_t bucket, wintop_bucket, new_bucket;
+	uint32_t bkt_diff, bkt_cnt;
+	uint64_t top_seq = 0, bit = 0;
+
+	odp_spinlock_lock(&ipsec_sa->hot.in.lock);
+
+	top_seq = odp_atomic_load_u64(&ipsec_sa->hot.in.wintop_seq);
+	if ((seq + ipsec_sa->ar.win_size) <= (top_seq & 0xffffffff))
```

Comment:
The anding with UINT32_MAX does not seem to serve any purpose, so it could be removed.

Reply:
Agreed, will remove it also the variable will be changed to 32-bit